x/ccv/provider/keeper/misbehaviour.go

```diff
@@ -0,0 +1,166 @@
+package keeper
+
+import (
+	"fmt"
+
+	ibcclienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
+	ibctmtypes "github.com/cosmos/ibc-go/v7/modules/light-clients/07-tendermint"
+
+	errorsmod "cosmossdk.io/errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	tmtypes "github.com/cometbft/cometbft/types"
+
+	"github.com/cosmos/interchain-security/v3/x/ccv/provider/types"
+)
+
+// HandleConsumerMisbehaviour checks if the given IBC misbehaviour corresponds to an equivocation light client attack,
+// and in this case, jails the Byzantine validators
+func (k Keeper) HandleConsumerMisbehaviour(ctx sdk.Context, misbehaviour ibctmtypes.Misbehaviour) error {
+	logger := k.Logger(ctx)
+
+	// Check that the misbehaviour is valid and that the client consensus states at trusted heights are within trusting period
+	if err := k.CheckMisbehaviour(ctx, misbehaviour); err != nil {
+		logger.Info("Misbehaviour rejected", err.Error())
+
+		return err
+	}
+
+	// Since the misbehaviour packet was received within the trusting period
+	// w.r.t to the trusted consensus states the infraction age
+	// isn't too old. see ibc-go/modules/light-clients/07-tendermint/types/misbehaviour_handle.go
+
+	// Get Byzantine validators from the conflicting headers
+	byzantineValidators, err := k.GetByzantineValidators(ctx, misbehaviour)
+	if err != nil {
+		return err
+	}
+
+	provAddrs := make([]types.ProviderConsAddress, len(byzantineValidators))
+
+	var errors []error
+	// slash, jail, and tombstone the Byzantine validators
+	for _, v := range byzantineValidators {
+		providerAddr := k.GetProviderAddrFromConsumerAddr(
+			ctx,
+			misbehaviour.Header1.Header.ChainID,
```

Comment:
Do we have a check somewhere that both `Header1` and `Header2` are for the same `ChainID`?
If not, could we have a case where headers are used from different chains?